packages/administration/src/Component/Config/Action/Builder/AbstractAction.php

```diff
@@ -81,27 +96,44 @@ public function setIcon(string $icon): self
     }
 
     /**
-     * Set CSS class that will be added to action button
+     * Set function that will determine if action should be displayed
      *
-     * @param string $cssClass
+     * @param \Closure(?object $entity): bool $function Function must return boolean value. If function returns false, action will not be displayed
      * @return $this
      */
-    public function setCssClass(string $cssClass): self
+    public function displayIf(Closure $function): self
     {
-        $this->cssClass = $cssClass;
+        $this->displayIf = $function;
 
         return $this;
     }
 
     /**
-     * Set function that will determine if action should be displayed
+     * Set attribute that will be passed to the template. This can be used to set for example `data-` attributes or change CSS classes.
+     * If value is null, attribute will be removed.
      *
-     * @param \Closure(?object $entity): bool $function Function must return boolean value. If function returns false, action will not be displayed
+     * @param string $name
+     * @param mixed $value
+     * @param bool $append If true, value will be appended to existing value. If false, existing value will be replaced.
      * @return $this
      */
-    public function displayIf(Closure $function): self
+    public function setAttribute(string $name, mixed $value, bool $append = false): self
     {
-        $this->displayIf = $function;
+        if (in_array($name, $this->forbiddenAttributes, true)) {
+            throw new InvalidArgumentException(sprintf('Attribute "%s" is forbidden to set.', $name));
```

Comment:
maybe it is worth mentioning something like "set this attribute using whateverFunction()"

Reply:
I can update the message to something like ... is forbidden to set. Look if any method sets this attribute... 

But `$this->forbiddenAttributes` can be updated by any child class so your suggestion could introduce additional overhead over configuring these values.

Reply:
Eventually, I got the idea. `$this->forbiddenAttributes` now accepts key as name of attribute and value as an additional error message.